ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4301,6 +4301,19 @@ public void recycleDirToCmPath(Path dataPath, boolean isPurge) throws HiveExcept
     }
   }
 
+  private static void deleteAndRename(FileSystem destFs, Path destFile, FileStatus srcStatus, Path destPath)
+          throws IOException {
+    if (destFs.exists(destFile)) {
+      // rename cannot overwrite non empty destination directory, so deleting the destination before renaming.
+      destFs.delete(destFile);
+      LOG.info("Deleting destination file" + destFile.toUri());
+    }
+    if(!destFs.rename(srcStatus.getPath(), destFile)) {
+      throw new IOException("rename for src path: " + srcStatus.getPath() + " to dest:"
+              + destPath + " returned false");
```

Comment:
destFile instead of destPath. Also message can be changed to "Failed to rename src: + srcStatus.getPath() + " to dest:"
               + destFile);

Reply:
The log is kept same as original code ..the rename might be of a file or directory

Reply:
OK, but dest should be destFile not destPath.